### PR TITLE
LPS-167141 - [Raylife AP] Create filter requests

### PR DESCRIPTION
**Pull request by pedroliveirav**
[Task: LPS-167141](https://issues.liferay.com/browse/LPS-167141)

### CODE REVIEW

**Comment by pedroliveirav** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 1]
```diff
@@ -12,176 +12,339 @@
...
+				return name
+					?.split(' ')
+					.map((product) => product.charAt(0))
+					.join('');
+			}
 
-			setPolicies(policiesList);
+			if (minNameLenght < name < maxNameLenght) {
+				return name.substring(0, 4);
+			}
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:112]:
```suggestion
			if (minProductNameLenght < name < maxProductNameLenght) {
				return productName.substring(0, 4);
			}
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 2]
```diff
@@ -12,176 +12,339 @@
...
 
-			return policiesList;
+			return name;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:114]:
```suggestion
			return productName;
```

**Review by joanagrego — APPROVED** [review]
Approved! I also added some suggestions :) 
Thank you @pedroliveirav :white_check_mark:

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 3]
```diff
@@ -12,176 +12,339 @@
...
-				}
-			});
+	const formatLabel = (labelColumnsFormatted, maxNameLenght) => {
+		const productAbbrevation = labelColumnsFormatted?.map((name) => {
+			if (name?.length > maxNameLenght) {
+				return name
+					?.split(' ')
+					.map((product) => product.charAt(0))
+					.join('');
+			}
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 1 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 2 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 4]
```diff
@@ -12,176 +12,341 @@
...
-		getPoliciesChartExpiringPolicies(
-			currentDateString[0],
-			currentDateString[1],
-			firstDayOfTheMonth,
-			nextMonthDate[0],
-			nextMonthDate[1],
-			nextMonthDate[2]
-		).then((results) => {
-			const totalFilteredPolicies = results?.data;
+	const options = [
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:56]: Can you better specify the name of this **options**? For example: filterOptions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 5]
```diff
@@ -12,176 +12,341 @@
...
+					currentDateString[2],
+					lastDateNextThreeMonth[0],
+					lastDateNextThreeMonth[1],
+					lastDateNextThreeMonth[2]
+				).then((results) => {
+					const policiesList = [];
+
+					const totalFilteredPolicies = results?.data;
+
+					totalFilteredPolicies?.items.map((policy) => {
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:215]: Items also need the '**?**'. Put it for everyone.

**Review by eliasantosf — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 4 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 5 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

**Review by eliasantosf — APPROVED** [review]
LGTM :fire: Thanks @pedroliveirav

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**